### ISSUE TITLE
Allowing any case style on all string built ins 

### ISSUE THREAD

**Issue by Al3xlol** [pynecone-io/pynecone#420]
pc.button(
    pc.icon(tag="MoonIcon"),
    on_click=pc.toggle_color_mode,
)
has same behavior as pc.button(
    pc.icon(tag="moonicon"),
    on_click=pc.toggle_color_mode,
)
so make it non case sensitive by using .lower() thanks and make sure to have this for any strings related to Pynecone

**Comment by elyanah-aco** [comment]
I'd like to work on this. Just need to know what specific components need to be changed under this

**Comment by Alek99** [comment]
Hi thanks for taking a look at this, I think a good place to start is the icon component. Feel free to join the discord if you need any help or guidance, or post follow ups here

**Comment by Al3xlol** [comment]
> Hi thanks for taking a look at this, I think a good place to start is the icon component. Feel free to join the discord if you need any help or guidance, or post follow ups here

^ yea icon components and other places in code base where strings are used I feel non case sensitive should be used

**Comment by elyanah-aco** [comment]
This was a little more complicated to solve than I actually thought. The below code addition for `icon.py` works in testing but doesn't feel rigorous enough. What do you think?
```
 if "tag" not in props.keys():
            raise AttributeError("Missing 'tag' keyword-argument for Icon")
        else:
            tag_keywords = [
                "add", "allowed", "arrow", "back", "circle", "clock", "close", "down", "forward",
                "handle", "icon", "left", "link", "off", "outline", "right", "sign", "two", "up"
            ]
            props['tag'] = props['tag'].title()
            for keyword in tag_keywords:
                props['tag'] = re.sub(keyword, keyword.title(), props['tag'])
```

**Comment by masenf** [comment]
Seems fixed via #553 unless there are other areas that are unaddressed.

```python
rx.icon(tag="RePeaT"),
```

is rendering correctly
